### PR TITLE
DESENG-237 Update CHES endpoints

### PR DESCRIPTION
**Pull request by Baelx**

### CODE REVIEW

File: api/helpers/email.js [hunk 1]
```diff
@@ -50,33 +64,23 @@ exports.sendConfirmEmail = async function ( projectName, email, confirmKey) {
...
-            {
-                headers: {
-                    'content-type': 'application/x-www-form-urlencoded'
-                }
-            }
-        );
-        if (getOKRes && getOKRes.data && getOKRes.data.access_token) {
+        const emailToken = await getEmailToken();
+
+        if (emailToken && emailToken.data && emailToken.data.access_token) {
```
Comment (ShawnTurple) [inline api/helpers/email.js:69]:
```suggestion
        if (emailToken?.data?.access_token) {
```

File: api/helpers/email.js [hunk 2]
```diff
@@ -109,33 +113,23 @@ exports.sendWelcomeEmail = async function (projectName, email) {
...
             // Send the welcome email
             await axios.post(
```
Comment (ShawnTurple) [inline api/helpers/email.js:120]: isn't line 71 axios call same as this line, wondering if we could combine these calls, similar to what you did with the getEmailToken function

**Review by ShawnTurple — APPROVED** [review]

**Comment by Baelx** [comment]
@ShawnTurple Thanks for the quick review. I have to make some additions to this PR and will consider your suggestions in terms of DRY code

File: api/helpers/email.js (hunk 2 above)
Comment (Baelx) [reply api/helpers/email.js:120]: After looking into this - I don't think there's an easy way to trim this further. Basically each email HTTP request requires an auth token. We currently request a token for each request before making it and as such we need to check that the token was successfully received beforehand.

Ideally we'd set up a request middleware that gets the token and saves it to memory(and refresh it as needed, upon expiry) so we don't to a token request each time we need to use the email API.

File: api/helpers/email.js (hunk 1 above)
Comment (Baelx) [reply api/helpers/email.js:69]: Unfortunately, the app uses Node 10, which doesn't support the optional chaining feature.

**Review by ShawnTurple — APPROVED** [review]

**Pull request merged**